cirq/vis/heatmap.py

```diff
@@ -52,220 +35,318 @@ def relative_luminance(color: np.ndarray) -> float:
+
+        colormap: Matplotlib colormap, default = viridis
+        vmin, vmax: colormap scaling floats, default = None
         """
-        # Fail fast if float() fails.
-        # Keep the original value object for annotation.
-        self.value_map = {qubit: (float(value), value) for qubit, value in value_map.items()}
-        return self
-
-    def set_annotation_map(
-        self, annot_map: Mapping[QubitCoordinate, str], **text_options: str
-    ) -> 'Heatmap':
-        """Sets the annotation text for each qubit.
+        self._value_map: Mapping[QubitTuple, SupportsFloat] = value_map
+        self._validate_kwargs(kwargs)
+        if '_config' not in self.__dict__:
+            self._config: Dict[str, Any] = {}
+        self._config.update(
+            {
+                "plot_colorbar": True,
+                "colorbar_position": "right",
+                "colorbar_size": "5%",
+                "colorbar_pad": "2%",
+                "colormap": "viridis",
+                "annotation_format": ".2g",
+            }
+        )
+        self._config.update(kwargs)
+
+    def _extra_valid_kwargs(self) -> List[str]:
+        return []
+
+    def _validate_kwargs(self, kwargs):
+        valid_colorbar_kwargs = [
+            "plot_colorbar",
+            "colorbar_position",
+            "colorbar_size",
+            "colorbar_pad",
+            "colorbar_options",
+        ]
+        valid_colormap_kwargs = [
+            "colormap",
+            "vmin",
+            "vmax",
+        ]
+        valid_heatmap_kwargs = [
+            "title",
+            "annotation_map",
+            "annotation_text_kwargs",
+            "annotation_format",
+        ]
+        valid_kwargs = (
+            valid_colorbar_kwargs
+            + valid_colormap_kwargs
+            + valid_heatmap_kwargs
+            + self._extra_valid_kwargs()
+        )
+        if any([k not in valid_kwargs for k in kwargs]):
+            invalid_args = ", ".join([k for k in kwargs if k not in valid_kwargs])
+            raise ValueError(f"Received invalid argument(s): {invalid_args}")
+
+    def update_config(self, **kwargs):
+        """Add/Modify **kwargs parameters passed during initialisation."""
+        self._validate_kwargs(kwargs)
+        self._config.update(kwargs)
+
+    def _qubits_to_polygon(self, qubits: QubitTuple) -> Tuple[Polygon, Point]:
+        qubit = qubits[0]
+        x, y = map(float, (qubit.row, qubit.col))
+        return (
+            [
+                (y - 0.5, x - 0.5),
+                (y - 0.5, x + 0.5),
+                (y + 0.5, x + 0.5),
+                (y + 0.5, x - 0.5),
+            ],
+            Point(y, x),
+        )
 
-        Note that set_annotation_map() and set_annotation_format()
-        both sets the annotation map to be used. Whichever is called later wins.
+    def _get_annotation_value(self, key, value):
+        if self._config.get('annotation_map', None):
```

Comment:
no need to define None as a default, that is the default value for dict.get()

Reply:
done